File: myRecipeBookFE/src/app/models/recipe.ts
import {Category} from './category';
import {Ingredient} from './ingredient';

export class Recipe {
  public recipeId?: number;
  public name: string;
  public category: Category;
  public categoryId?: number;
  public description: string;
  public imagePath: string;
  public ingredients?: Ingredient[];
  public userId?: number;

  constructor(
    recipeId: number,
    name: string,
    category: Category,
    categoryId: number,
    description: string,
    imagePath: string,
    ingredients: Ingredient[],
    userId: number
  ) {
    this.recipeId = recipeId;
    this.name = name;
    this.category = category;
    this.categoryId = categoryId;
    this.description = description;
    this.imagePath = imagePath;
    this.ingredients = ingredients;
    this.userId = userId;
  }
}

File: myRecipesBook/src/main/java/com/ceylan/myRecipesBook/entity/concrete/Recipe.java
package com.ceylan.myRecipesBook.entity.concrete;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.util.List;

@Table(name = "recipes")
@Data
@Entity
@NoArgsConstructor
@AllArgsConstructor
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Recipe {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "recipe_id")
    private int recipeId;

    @Column(name = "name")
    private String name;

    @Column(name = "description")
    private String description;

    @Column(name = "imagePath")
    private String imagePath;

    @ManyToOne(fetch = FetchType.LAZY, targetEntity = Category.class)
    @JoinColumn(name = "category_id")
    private Category category;

    @JsonIgnore
    @OneToMany(mappedBy = "recipe")
    private List<Ingredient> ingredients;

    @ManyToOne(fetch = FetchType.LAZY, targetEntity = User.class)
    @JoinColumn(name = "user_id")
    private User user;


}